server.cpp

```diff
@@ -305,26 +326,57 @@ void FileGopherServer::start(int n_connections){
  * @param route The path to route.
  * @param content The contents of route.
  */
-void FileGopherServer::add_route(string route, string content) {
+void FileGopherServer::add_route(string route, string content)
+{
     routes[route] = content;
 }
 
-
 // Driver function.
-int main(int argc, char *argv[]){
-    // Print Usage.
-    if(argc != 2){
-        printf("USAGE: \n\tserver <directory-to-serve>\n\n");
-        printf("Where <directory-to-serve> root directory of server.\n");
-        return 0;
+int main(int argc, char *argv[])
+{
+    // variables needed to be captured from command line options
+
+    string directory = "."; // defaults to cwd
+    int port = 70;          // default port value of 70
+    bool show_hidden_files = false;
+
+    // Declare the supported options.
+    po::options_description desc("Allowed options");
+    desc.add_options()("directory,d", po::value<string>(), "the directory to 
```

Comment:
Did you compile this and check?

I think the non-terminated string at the end should throw an error.